submarine-server/server-core/src/main/java/org/apache/submarine/server/environment/EnvironmentManager.java

```diff
@@ -204,9 +204,31 @@ public Environment getEnvironment(String name)
    */
   public List<Environment> listEnvironments(String status)
       throws SubmarineRuntimeException {
-    List<Environment> environmentList =
+    List<Environment> envs =
         new ArrayList<Environment>(cachedEnvironments.values());
-    return environmentList;
+
+    if (envs.size() != 0) {
+      return envs;
+    }
+    try (SqlSession sqlSession = MyBatisUtil.getSqlSession()) {
+      EnvironmentMapper environmentMapper =
+            sqlSession.getMapper(EnvironmentMapper.class);
+      List<EnvironmentEntity> environmentEntitys = environmentMapper.selectAll();
+      for (EnvironmentEntity environmentEntity : environmentEntitys) {
+        if (environmentEntity != null) {
+          Environment env = new Environment();
+          env.setEnvironmentSpec(new Gson().fromJson(
+                environmentEntity.getEnvironmentSpec(), EnvironmentSpec.class));
+          envs.add(env);
+          cachedEnvironments.put(env.getEnvironmentSpec().getName(), env);
+        }
+      }
+    } catch (Exception e) {
+      LOG.error(e.getMessage(), e);
+      throw new SubmarineRuntimeException(Status.BAD_REQUEST.getStatusCode(),
+            "Unable to get the environment details.");
```

Comment:
Maybe it would be better to change to `Unable to get the environment list.` ?

Reply:
I will rewrite this description :)